### DISCUSSION TITLE
Amazon S3 Download Image without it been Public

### DISCUSSION THREAD

**Discussion by SuperlativeEntity** [spatie/laravel-medialibrary#1994] [Q&A]
Your packages are amazing by the way, real game-changers and pure pleasure to use!

So I'm able to upload an image to S3. but I couldn't get it to render using {{ $media }}

I then added a policy to Amazon S3 to be able to download the image, which now rendered.

```
{
    "Version": "2012-10-17",
    "Id": "Policy1597242711944",
    "Statement": [
        {
            "Sid": "Stmt1597242710724",
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject",
                "s3:GetObjectAcl"
            ],
            "Resource": "arn:aws:s3:::za.co.bucket_name/*"
        }
    ]
}
```

This, however, makes the image publically accessible.

I then went onto the Laravel discord and they said I need to use a signed URL and provided this example.

`$mediaItems[0]->getTemporaryUrl(Carbon::now()->addMinutes(5), 'thumb');`

But I'm not converting the image, I just want to access it without a public policy.

Please accept my apologies in advance if this is a S3 setup issue and not a part of the package.

**Comment by SuperlativeEntity** [comment]
Just did this...

Then pass the image in the view to display it

```
    public function getTemporaryUrl(string $fullFilePath,int $numberOfMinutes,string $disk)
    {
        return Storage::disk($disk)->temporaryUrl($fullFilePath,now()->addMinutes($numberOfMinutes));
    }
```

**Comment by romansidorov** [comment]
Hi

If you need to download it (as you typed in the header above) you can return an object: https://spatie.be/docs/laravel-medialibrary/v7/downloading-media/downloading-a-single-file
